### ISSUE TITLE
Setting a Poetry constraint causes lockfile update to fail

### ISSUE THREAD

**Issue by WarpRat** [renovatebot/renovate#19972]
### How are you running Renovate?

Mend Renovate hosted app on github.com

### If you're self-hosting Renovate, tell us what version of Renovate you run.

Present on self-hosted and github app

### If you're self-hosting Renovate, select which platform you are using.

gitlab.com

### If you're self-hosting Renovate, tell us what version of the platform you run.

Also observed on gitlab.com 15.8

### Was this something which used to work for you, and then stopped?

I never saw this working

### Describe the bug

When setting a python and poetry constraint, the poetry lock update command fails because renovate attempts to pip install poetry using the wrong format for pip version constraints. See the repro here: https://github.com/WarpRat/renovate-poetry-contraint-repro/pull/6

### Relevant debug logs

<details><summary>Logs</summary>

```
DEBUG: Failed to update poetry.lock file(branch="renovate/pylint-2.x")
{
  "err": {
    "name": "ExecError",
    "cmd": "/bin/sh -c docker run --rm --name=renovate_sidecar --label=renovate_child -v \"/mnt/renovate/gh/WarpRat/renovate-poetry-contraint-repro\":\"/mnt/renovate/gh/WarpRat/renovate-poetry-contraint-repro\" -v \"/tmp/renovate-cache\":\"/tmp/renovate-cache\" -v \"/tmp/containerbase\":\"/tmp/containerbase\" -e PIP_CACHE_DIR -e BUILDPACK_CACHE_DIR -e CONTAINERBASE_CACHE_DIR -w \"/mnt/renovate/gh/WarpRat/renovate-poetry-contraint-repro\" docker.io/renovate/sidecar bash -l -c \"install-tool python 3.9.9 && pip install --user poetry1.3.2 && poetry update --lock --no-interaction pylint\"",
    "stderr": "ERROR: Could not find a version that satisfies the requirement poetry1.3.2 (from versions: none)\nERROR: No matching distribution found for poetry1.3.2\n",
    "stdout": "installing v2 tool python v3.9.9\nlinking tool python v3.9.9\nPython 3.9.9\npip 22.3.1 from /opt/buildpack/tools/python/3.9.9/lib/python3.9/site-packages/pip (python 3.9)\nInstalled v2 /usr/local/buildpack/tools/v2/python.sh in 24 seconds\nskip cleanup, not a docker build: 8d08c5ae1170\n",
    "options": {
      "cwd": "/mnt/renovate/gh/WarpRat/renovate-poetry-contraint-repro",
      "encoding": "utf-8",
      "env": {
        "PIP_CACHE_DIR": "/tmp/renovate-cache/others/pip",
        "HOME": "/home/ubuntu",
        "PATH": "/home/ubuntu/.local/bin:/home/ubuntu/bin:/opt/buildpack/tools/python/3.9.3/bin:/home/ubuntu/.npm-global/bin:/home/ubuntu/renovateapp/node_modules/.bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin",
        "LC_ALL": "C.UTF-8",
        "LANG": "C.UTF-8",
        "BUILDPACK_CACHE_DIR": "/tmp/containerbase",
        "CONTAINERBASE_CACHE_DIR": "/tmp/containerbase"
      },
      "maxBuffer": 10485760,
      "timeout": 900000
    },
    "exitCode": 1,
    "message": "Command failed: docker run --rm --name=renovate_sidecar --label=renovate_child -v \"/mnt/renovate/gh/WarpRat/renovate-poetry-contraint-repro\":\"/mnt/renovate/gh/WarpRat/renovate-poetry-contraint-repro\" -v \"/tmp/renovate-cache\":\"/tmp/renovate-cache\" -v \"/tmp/containerbase\":\"/tmp/containerbase\" -e PIP_CACHE_DIR -e BUILDPACK_CACHE_DIR -e CONTAINERBASE_CACHE_DIR -w \"/mnt/renovate/gh/WarpRat/renovate-poetry-contraint-repro\" docker.io/renovate/sidecar bash -l -c \"install-tool python 3.9.9 && pip install --user poetry1.3.2 && poetry update --lock --no-interaction pylint\"\nERROR: Could not find a version that satisfies the requirement poetry1.3.2 (from versions: none)\nERROR: No matching distribution found for poetry1.3.2\n",
    "stack": "ExecError: Command failed: docker run --rm --name=renovate_sidecar --label=renovate_child -v \"/mnt/renovate/gh/WarpRat/renovate-poetry-contraint-repro\":\"/mnt/renovate/gh/WarpRat/renovate-poetry-contraint-repro\" -v \"/tmp/renovate-cache\":\"/tmp/renovate-cache\" -v \"/tmp/containerbase\":\"/tmp/containerbase\" -e PIP_CACHE_DIR -e BUILDPACK_CACHE_DIR -e CONTAINERBASE_CACHE_DIR -w \"/mnt/renovate/gh/WarpRat/renovate-poetry-contraint-repro\" docker.io/renovate/sidecar bash -l -c \"install-tool python 3.9.9 && pip install --user poetry1.3.2 && poetry update --lock --no-interaction pylint\"\nERROR: Could not find a version that satisfies the requirement poetry1.3.2 (from versions: none)\nERROR: No matching distribution found for poetry1.3.2\n\n    at ChildProcess.<anonymous> (/home/ubuntu/renovateapp/node_modules/renovate/dist/util/exec/common.js:87:24)\n    at ChildProcess.emit (node:events:525:35)\n    at ChildProcess.emit (node:domain:489:12)\n    at Process.ChildProcess._handle.onexit (node:internal/child_process:293:12)"
  }
}
```

</details>

### Have you created a minimal reproduction repository?

I have linked to a minimal reproduction repository in the bug description

**Comment by mounirmesselmeni** [comment]
Any easy workaround for this?

I was stuck with poetry in gitlab dependabot as dependabot-core is used as library and they didn't achieve a release since December 2022.

I was so glad I find out about Renovate and now still issues with Poetry. 
Maybe I should blame myself choosing poetry 😆

**Comment by rarkins** [comment]
You can set your constraint like `==1.2.3` instead of `1.2.3`

**Comment by mounirmesselmeni** [comment]
@rarkins doesn't help if I have other conditional dependencies like "0.2.*", not convinced to change them all

**Comment by rarkins** [comment]
I mean in your Renovate config file. If you don't have one, then use one. Example of it being wrong is https://github.com/WarpRat/renovate-poetry-contraint-repro/blob/463f9f551f4832c0e8fad743e37ef18da54c6e0d/.gitlab/renovate.json#L1-L8

Change `1.3.2` to `==1.3.2` and it works

**Comment by mounirmesselmeni** [comment]
@rarkins Thank you.
I mixed things up, my issue appeared to be that if I don't define `GITHUB_COM_TOKEN` I got an error for generating the lock file of Poetry which was confusing and tricky to find out.
I was getting an error that Renovate got an error generating lock files for Poetry but not with much details, when I activated DEBUG logs I see the error related to "No tool releases found".

I suppose it's more user friendly if a proper error is triggered in such cases. But now it works great and I really appreciate your work and this great tool!

Thanks for your help!
